tools/realsense-viewer/realsense-viewer.cpp

```diff
@@ -205,7 +205,20 @@ bool refresh_devices(std::mutex& m,
                 if (device_models.size() == 0 &&
                     dev.supports(RS2_CAMERA_INFO_NAME) && std::string(dev.get_info(RS2_CAMERA_INFO_NAME)) != "Platform Camera" && std::string(dev.get_info(RS2_CAMERA_INFO_NAME)).find("IP Device") == std::string::npos)
                 {
-                    device_models.emplace_back(new device_model(dev, error_message, viewer_model));
+                    try
+                    {
+                        device_models.emplace_back(new device_model(dev, error_message, viewer_model));
+                    }
+                    catch (const std::exception& e)
+                    {
+                        auto dev_name_itr = std::find(begin(device_names), end(device_names), get_device_name(dev));
```

Comment:
Maybe we can add a LOG or something?
That we failed creating the new connected device ?

Reply:
sure!